3party/protobuf/CMakeLists.txt

```diff
@@ -1,43 +1,70 @@
 project(protobuf)
 
-set(SRC
-  config.h
-  protobuf/src/google/protobuf/arena.cc
-  protobuf/src/google/protobuf/extension_set.cc
-  protobuf/src/google/protobuf/generated_message_util.cc
-  protobuf/src/google/protobuf/io/coded_stream.cc
-  protobuf/src/google/protobuf/io/zero_copy_stream.cc
-  protobuf/src/google/protobuf/io/zero_copy_stream_impl_lite.cc
-  protobuf/src/google/protobuf/message_lite.cc
-  protobuf/src/google/protobuf/repeated_field.cc
-  protobuf/src/google/protobuf/stubs/atomicops_internals_x86_gcc.cc
-  protobuf/src/google/protobuf/stubs/bytestream.cc
-  protobuf/src/google/protobuf/stubs/common.cc
-  protobuf/src/google/protobuf/stubs/int128.cc
-  protobuf/src/google/protobuf/stubs/once.cc
-  protobuf/src/google/protobuf/stubs/status.cc
-  protobuf/src/google/protobuf/stubs/statusor.cc
-  protobuf/src/google/protobuf/stubs/stringpiece.cc
-  protobuf/src/google/protobuf/stubs/stringprintf.cc
-  protobuf/src/google/protobuf/stubs/structurally_valid.cc
-  protobuf/src/google/protobuf/stubs/strutil.cc
-  protobuf/src/google/protobuf/stubs/time.cc
-  protobuf/src/google/protobuf/wire_format_lite.cc
-)
+# Disable all compile-time warnings for protobuf
+add_compile_options(-w)
 
-add_library(${PROJECT_NAME} ${SRC})
+# protobuf config
+set(protobuf_LIB_NAME libprotobuf-lite)
+set(protobuf_VERSION "23.4")
```

Comment:
Documenting this can be challenging, as the documentation might become outdated with new releases. Normally, updating the version number should suffice. However, if there are alterations like the addition of a new library or changes in configuration, it will be necessary to carefully examine the modifications and adjust our configuration to match them.

Reply:
Am I correctly understanding that you have created this file manually to wrap protobuf sources? Why the bundled [CMakeLists.txt](https://github.com/protocolbuffers/protobuf/blob/main/CMakeLists.txt) can't be customized and used via add_subdirectory?

Reply:
It's not possible because protobuf has conflicting target's name. It was already discussed here: https://github.com/organicmaps/organicmaps/pull/5188#issuecomment-1556212319

Reply:
Thanks for reminding me about that comment. So that is the real reason for creating a copy of CMakeLists.txt manually.

1. Can FindPackage + alias approach work here instead of add_subdirectory with a manual copy of CMakeLists.txt?
2. Can a lower, more recent 3x version without a conflicting `base` target be considered instead?

Reply:
If I correctly understood, 3.21.9 version (21.x) was building fine, right @Ferenc- ? Was it the latest one without abseil and conflicting targets? https://github.com/organicmaps/organicmaps/pull/3932